### DISCUSSION TITLE
Query Builder Columns/Measure - select all

### DISCUSSION THREAD

**Discussion by albertan352652** [DaxStudio/DaxStudio#619] [Q&A]
Is there a way to bring all columns/measures selected to query builder or do I need to select them one by one and bring them? Thanks for your help

**Comment by dgosbell** [comment]
I'm not entirely sure what you are asking here. The current Microsoft TreeView control that we are using in DAX Studio does not support multiple selections so the concept of "all selected columns/measures" does not currently exist. If you are asking for a multi-selection option in the treeview unfortunately that is not just a simple setting that can be changed.

One alternative workflow is that you can use the [keyboard shortcuts](https://daxstudio.org/documentation/features/query-builder/#keyboard-shortcuts) either in combination with mouse clicks or using the search box (I usually change the setting in the options to keep the search open all the time) This _should_ enable you to very quickly build up a query

**Reply by albertan352652** [reply]
Thanks I understand that If I have a table with 100 columns for example and I want to perform Query Editor task with putting a filter on 1 column and adding remaining 99 columns then I will have  to select those 99 columns one by one.
I.e. I cannot select fist field, then press Shift Key and last key so that it will make a whole selection. That would have been nice to have.

**Reply by dgosbell** [reply]
You are correct, there is currently no way to do that sort of thing easily with the query builder. I might do some investigation and see if I can do something with dragging a whole table to the query builder and either expanding it into a list of columns at run time or letting the user expand the columns in the designer.

**Reply by albertan352652** [reply]
Hello, just wanted to mark your answer as final, unless you have any update for possibility to have it easily done with the query builder? Thanks

**Reply by dgosbell** [reply]
It can be done with the query builder. I've added an item here #1135 to track this
